[Kaka0 adult] 장바구니 리스트 get method 구현

hyo-jin-jeong opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 장바구니 리스트 get method 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 장바구니 리스트 get method 구현

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 예외처리 해야 할 부분이 있을까요?
- user 장바구니가 없으면 생성해서 빈리스트를 보내주어야 하고, 있으면 get 해야해서 get_or_create 메소드를 사용하였습니다. 
request.user.order_set해서 장바구니가 있는지 확인하고 create를 하는 방식이 더 좋을까요?
<img width="417" alt="스크린샷 2021-06-15 오전 10 58 33" src="https://user-images.githubusercontent.com/55984573/121981450-a2001600-cdc8-11eb-9234-a9afe31736ed.png">

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hyo-jin-jeong committed:
Add: 장바구니 리스트 get method 추가

hyo-jin-jeong committed:
merge main

hyo-jin-jeong committed:
Add: 장바구니 리스트 get method 추가

hyo-jin-jeong committed:
Modify: 장바구니 리스트 get method 구현

Kang-kyunghun commented on db_uploader.py:
```diff
@@ -0,0 +1,71 @@
+    if options.delete:
+       delete_table_records()
+    
+    if options.create:
+        create_table_records()
```
db_uploader 파일은 제거하고 gitignore에 추가해 주세요.

hyo-jin-jeong replied:
수정했습니다!

Kang-kyunghun commented on orders/urls.py:
```diff
@@ -5,4 +5,5 @@
 urlpatterns = [
     path('/order-item', BasketView.as_view()),
     path('/order-items/<int:order_item>', BasketView.as_view()),
+    path('/order-items', BasketView.as_view()),
```
위에 똑같은 endpoint가 존재하는 것 같은데 확인해 주세요.

hyo-jin-jeong replied:
수정했습니다!

Kang-kyunghun commented on orders/views.py:
```diff
@@ -57,4 +57,25 @@ def delete(self, request, order_item):
+
+    @login_decorator
+    def get(self, request):
+        order, created = Order.objects.get_or_create(
+            user            = request.user,
+            order_status_id = OrderStatus.BASKET
+        )
```
조회 기능인데, create를 하는 이유가 무엇일까요?

hyo-jin-jeong replied:
수정했습니다!

hyo-jin-jeong committed:
Modify: BasketView 변경

Kang-kyunghun commented on orders/views.py:
```diff
@@ -57,4 +57,25 @@ def delete(self, request, order_item):
+                'stock'        : order_item.product.stock,
+                'image_url'    : order_item.product.imageurl_set.order_by('id')[0],
+                'selected'     : order_item.selected
+            }for order_item in order.orderitem_set.all()]
```
```suggestion
            }for order_item in OrderItem.objects.filter(조건)]
```
결국 저희는 OrderItem 중에 장바구니 조건을 만족하는 제품들을 보여주는 것이므로 위와 같이 수정하면 좋을 것 같아요

hyo-jin-jeong replied:
수정했습니다!

Kang-kyunghun commented on orders/views.py:
```diff
@@ -57,4 +57,25 @@ def delete(self, request, order_item):
+            }for order_item in order.orderitem_set.all()]
+
+        return JsonResponse({'message':'SUCCESS', 'items_in_cart':order_items}, status=200)
+
+
```
공백 제거해 주세요.

hyo-jin-jeong replied:
수정했습니다!

hyo-jin-jeong committed:
Refactor: "멘토님 리뷰 반영"
- db_uploader 파일 제거
- endpoint 변경
- create -> get method 변경
- all() -> filter() 변경
- 공백 제거

hyo-jin-jeong committed:
Modify: img url 가져오기

Kang-kyunghun commented on orders/views.py:
```diff
@@ -57,4 +57,21 @@ def delete(self, request, order_item):
+                'count'        : order_item.count,
+                'price'        : order_item.product.price,
+                'stock'        : order_item.product.stock,
+                'image_url'    : order_item.product.imageurl_set.order_by('id')[0].url,
```
대표이미지(썸네일) 같은 경우는 product에 컬럼을 만들어서 사용하는 것도 좋은 방법 중 하나 입니다.
그러면 대표 이미지의 정보를 얻기 위해서 역참조를 안 하고 바로 product 테이블에서 가져 올 수 있습니다.
이거는 일단 참고 사항이니 당장 수정하지 않으셔도 됩니다.

Kang-kyunghun commented:
LGTM~!!
